### ISSUE TITLE
Building for x86_64-unknown-linux-gnu on Windows host fails on Rust 1.50

### ISSUE THREAD

**Issue by vilcans** [cross-rs/cross#527]
After upgrading to Rust 1.50, I can no longer build for Linux from a Windows host.

It looks like Cross tries to run the Linux toolchain under Windows and then fails because of some conflict with the documentation:

```
$ cargo init crosstest
     Created binary (application) package
$ cd crosstest/
$ winpty cross build --target x86_64-unknown-linux-gnu
warning: toolchain '1.50-x86_64-unknown-linux-gnu' may not be able to run on this system.
warning: If you meant to build software to target that platform, perhaps try `rustup target add x86_64-unknown-linux-gnu` instead?
info: syncing channel updates for '1.50-x86_64-unknown-linux-gnu'
info: latest update on 2021-02-11, rust version 1.50.0 (cb75ad5db 2021-02-10)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'rust-docs'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: using up to 500.0 MiB of RAM to unpack components
info: installing component 'clippy'
info: installing component 'rust-docs'
info: rolling back changes
error: failed to install component: 'rust-docs-x86_64-unknown-linux-gnu', detected conflict: '"share/doc/rust/html/std/keyword.self.html"'
error: couldn't install toolchain `1.50-x86_64-unknown-linux-gnu`
caused by: `"rustup" "toolchain" "add" "1.50-x86_64-unknown-linux-gnu"` failed with exit code: Some(1)
note: run with `RUST_BACKTRACE=1` for a backtrace
```

That looks like it does not start a Docker container, and tries to install the tools on the host machine.

In Rust 1.49, it works:

```
$ rustup default 1.49
info: using existing install for '1.49-x86_64-pc-windows-msvc'
info: default toolchain set to '1.49-x86_64-pc-windows-msvc'

  1.49-x86_64-pc-windows-msvc unchanged - rustc 1.49.0 (e1884a8e3 2020-12-29)

$ winpty cross build --target x86_64-unknown-linux-gnu
   Compiling crosstest v0.1.0 (/project)
    Finished dev [unoptimized + debuginfo] target(s) in 14.22s
```

(I'm running MSYS2, where the winpty command is required to run Cross with Docker. On the failed command it wasn't needed because it didn't use Docker in the first place.)

**Comment by vilcans** [comment]
A workaround is to set the minimal profile before building:

`rustup set profile minimal`

Then rustup won't try to install documentation, and will not fail at that step. Still strange that this changed between 1.49 and 1.50, but it may not be a bug in Cross. #533 seems to be the same.

**Comment by JayKickliter** [comment]
>  but it may not be a bug in Cross

I don't think it is, but a workaround may be necessary in cross (if that's even possible).

Relevant rustup comment: https://github.com/rust-lang/rustup/issues/2661#issuecomment-783717159

**Comment by rbtcollins** [comment]
Hi, wearing a rustup-maintainer hat I would argue this is a bug in cross: since cross knows it is doing a generally unusal thing (installing toolchains that cannot work on the host, and doing so solely for automation), cross should override the profile on the toolchain install request to rustup - always installing new toolchains with the minimal profile. This will make cross faster, use less network resources, and avoid this conflict.

**Comment by cbc02009** [comment]
This also happens to me on Windows for embedded (thumb*) builds. @vilcans solution fixed it.

**Comment by KB5201314** [comment]
I think the problem is caused by these lines of code:
https://github.com/rust-embedded/cross/blob/3d4004cf62de81c0bb9fffcae045cbeed449211d/src/rustc.rs#L61-L65
As @rbtcollins said, if the toolchain being used on the host is not `*-x86_64-unknown-linux-gnu`, cross will try to download one, and then bind the corresponding `sysroot` directory into the docker container.
https://github.com/rust-embedded/cross/blob/efb6e591e43a29965fbb6b924c4ad0e8ce37164d/src/docker.rs#L191

**Comment by Alexhuszagh** [comment]
This should be fixed as of e05540fbf329a446aa4cab47ced9269c3273f640. Please let me know if the issue still reproduces: I've tested this fairly regularly on other hosts (for example, `x86_64-pc-windows-msvc`)  so I don't believe this is still relevant.
